opentelemetry-sdk/src/logs/log_processor.rs

```diff
@@ -994,4 +998,250 @@ mod tests {
+
+            // This will panic because an tokio async operation within exporter without a runtime.
+            processor.emit(&mut record, &instrumentation);
+        });
+
+        // Verify that the panic occurred and check the panic message for the absence of a Tokio runtime
+        assert!(
+            result.is_err(),
+            "The test should fail due to missing Tokio runtime, but it did not."
+        );
+        let panic_payload = result.unwrap_err();
+        let panic_message = panic_payload
+            .downcast_ref::<String>()
+            .map(|s| s.as_str())
+            .or_else(|| panic_payload.downcast_ref::<&str>().copied())
+            .unwrap_or("No panic message");
+
+        assert!(
+            panic_message.contains("no reactor running")
+                || panic_message.contains("must be called from the context of a Tokio 1.x runtime"),
+            "Expected panic message about missing Tokio runtime, but got: {}",
+            panic_message
+        );
+    }
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 4)]
+    #[ignore]
+    // This test demonstrates a potential deadlock scenario in a multi-threaded Tokio runtime.
+    // It spawns Tokio tasks equal to the number of runtime worker threads (4) to emit log events.
+    // Each task attempts to acquire a mutex on the SimpleLogProcessor. Only one task obtains the lock,
+    // while the others are blocked, waiting for its release.
+    //
+    // The task holding the lock invokes the LogExporterThatRequiresTokio, which performs an
+    // asynchronous operation (e.g., network I/O simulated by `tokio::sleep`). This operation
+    // requires yielding control back to the Tokio runtime to make progress.
+    //
+    // However, all worker threads are occupied:
+    // - One thread is executing the async exporter operation
+    // - Three threads are blocked waiting for the mutex
+    //
+    // This leads to a deadlock as there are no available threads to drive the async operation
+    // to completion, preventing the mutex from being released. Consequently, neither the blocked
+    // tasks nor the exporter can proceed.
+    async fn test_simple_processor_async_exporter_with_all_runtime_worker_threads_blocked() {
+        let exporter = LogExporterThatRequiresTokio::new();
+        let processor = Arc::new(Mutex::new(SimpleLogProcessor::new(Box::new(
+            exporter.clone(),
+        ))));
+
+        let concurrent_emit = 4; // number of worker threads
+
+        let mut handles = vec![];
+        // try send `concurrent_emit` events concurrently
+        for _ in 0..concurrent_emit {
+            let processor_clone = Arc::clone(&processor);
+            let handle = tokio::spawn(async move {
+                let mut record: LogRecord = Default::default();
+                let instrumentation: InstrumentationLibrary = Default::default();
+                processor_clone
+                    .lock()
```

Comment:
Remove this mutex around the SimpleLogProcessor? The `processor.emit` code already takes a lock before calling `exporter.export` so this does not seem to add any value on top of that.

Reply:
Very good point. Have removed this now.